docs/hvordan-utvikle-maler.adoc

```diff
@@ -0,0 +1,550 @@
+            textExpr(
+                Bokmal to "Du får utbetalt ".expr() + beloep.format() + " Kroner per måned før skatt",
+                Nynorsk to ...,
+                English to ...,
+            )
+        }
+    }
+}
+----
+
+Det finnes 3 typer fraser, hvor de kan bli inkludert ulike steder:
+TextOnlyPhrase er ren tekst, og kan f.eks brukes inne i celler og lister.
+ParagraphPhrase kan inneholde elementer som kan brukes inne i ett avsnitt. F.eks tabeller, tekster, lister osv.
+OutlinePhrase kan inneholde elementer som kan skrives i outline. F.eks title1, title2, paragraph.
+
+== Vedlegg
+Brev har oftest vedlegg som kommer på egne ark i pdf etter ett brev og inkluderes i vedlegg listen.
+For utvikling av maler trenger man ikke å bry seg så veldig mye om hva som skjer i bakgrunnen.
+
+Ett vedlegg defineres slik og legges på ett fornuftig sted i vedlegg pakken:
+[%nowrap, kotlin, ]
+----
+@TemplateModelHelpers
+val eksempelVedlegg = createAttachment<LangBokmalNynorskEnglish, EksempelVedleggDto>(
+    // På samme måte som brev kan vedlegg kreve data for å produseres.
+    // Vanlig mønster for disse data klassene er at de inkluderes i brev som bruker vedlegget, så alle bruker samme data klasse for vedlegget.
+    // Dette gjør det enkelt å gjøre endringer i ett vedlegg som brukes på tvers av mange brev
+
+    // På samme måte som ett brev må vedlegg ha en tittel
+    title = newText(
+        Bokmal to "Dine rettigheter og mulighet til å klage",
+        Nynorsk to "Rettane dine og høve til å klage",
+        English to "Your rights and how to appeal"
+    ),
+    // Setter om informasjon om bruker/verge og saksnummer skal vises på samme måte som hoved-brevet i vedlegget.
+    includeSakspart = false,
+) {
+    // Innhold defineres på lik måte som outline
+}
+----
+
+=== Inkludering av vedlegg i maler
+Vedlegg kan inkluderes utenfor outline ved å bruke enten includeAttachment eller includeAttachmentIfNotNull
+
+==== includeAttachment
+[%nowrap, kotlin, ]
+----
+{
+    outline {...}
+    includeAttachment(
+        template = eksempelVedlegg,
+        attachmentData = eksempelVedleggData
+        predicate = skalHaVedlegget /*logikk som styrer om vedlegget skal produseres.*/
+    )
+}
+----
+==== includeAttachmentIfNotNull
+includeAttachmentIfNotNull inkluderer vedlegget med dataene om dataene ikke er null. Typisk bruk for denne funksjonen
+er om det at vedlegget vises henger sammen med at vi har dataene. Da kan vi også garantere i vedlegget at dataene er satt.
+[%nowrap, kotlin, ]
+----
+{
+    outline {...}
+    includeAttachmentIfNotNull(
+        template = eksempelVedlegg,
+        attachmentData = eksempelVedleggData /* hvor denne kan være null */
+    )
+}
+----
+
+
+
+== Eksempel på en enkel mal
+[%nowrap, kotlin, ]
+----
+@TemplateModelHelpers // Annotasjon som gjør at malen blir plukket opp av en kode-generator. Kode-generatoren brukes bare for å lage utility-funksjoner så vi slipper å være så verbose i brevmalen
```

Comment:
Forslag:
// Annotasjon som gjør at malen blir plukket opp av en kode-generator. Det vil da genereres extension functions  og properties basert på data-klassen som er angitt for malen som gir deg lett tilgang til feltene i en dataklasse i scopet til malen.